### PR TITLE
Remove unlock mechanism

### PR DESCRIPTION
**Pull request by pandadefi**
## Description

With the migration mechanism we don't need the unlock mechanism.

Fixes # (issue)

## Checklist

- [x] I have run vyper and solidity linting
- [x] I have run the tests on my machine
- [x] I have followed commitlint guidelines
- [x] I have rebased my changes to the latest version of the main branch

### CODE REVIEW

File: contracts/VotingEscrow.vy [hunk 1]
```diff
@@ -433,7 +440,6 @@ def deposit_for(_addr: address, _value: uint256):
     @param _addr User's wallet address
     @param _value Amount to add to user's lock
     """
-    assert(self.unlocked == False) # dev: no more lock
```
Comment (rareweasel) [inline contracts/VotingEscrow.vy:436]: Question:

I see some LoC deleted with the code `assert(self.unlocked == False)`

Shouldn't use `self.migrated` instead?

Comment (rareweasel) [inline contracts/VotingEscrow.vy:436]: Question:

Shouldn't add `self.migrated` here and [here](https://github.com/yearn/veYFI/pull/104/files#diff-244a97f574a321ee15a9bbcde95f07842637249f93f58dec8c39f6587efd272dL453)?

File: contracts/VotingEscrow.vy [hunk 2]
```diff
@@ -597,6 +593,28 @@ def force_withdraw():
...
+    ERC20(self.token).approve(self.next_ve_contract, value)
+    Migrator(self.next_ve_contract).migrateLock(msg.sender, value)
+
+    old_locked: LockedBalance = _locked
+    _locked.end = 0
+    _locked.amount = 0
+    self.locked[msg.sender] = _locked
+    supply_before: uint256 = self.supply
+    self.supply = supply_before - value
+    self._checkpoint(msg.sender, old_locked, _locked)
```
Comment (rareweasel) [inline contracts/VotingEscrow.vy:614]: Question:

Should we clear the `next_ve_contract` value after migrating?

Comment (rareweasel) [inline contracts/VotingEscrow.vy:614]: Question:

Should we clear the `next_ve_contract` value after migrating?

File: contracts/test/NextVe.sol [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.12;
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+contract NextVe {
```
Comment (rareweasel) [inline contracts/test/NextVe.sol:6]: Suggestion:

I would extend from an interface.

Comment (rareweasel) [inline contracts/test/NextVe.sol:6]: Suggestion:

I would extend from an interface.

File: tests/functional/test_voting_escrow.py [hunk 4]
```diff
@@ -373,3 +371,19 @@ def test_apply_transfer_ownership(ve_yfi, accounts):
...
+    yfi.mint(panda, amount, {"from": panda})
+    yfi.approve(ve_yfi.address, amount, {"from": panda})
+
+    ve_yfi.create_lock(amount, chain[-1].timestamp + 2 * WEEK, {"from": panda})
+    next_ve = gov.deploy(NextVe, yfi)
+    ve_yfi.set_next_ve_contract(next_ve)
+    ve_yfi.migrate({"from": panda})
+    assert ve_yfi.balanceOf(panda) == 0
+
+    assert yfi.balanceOf(next_ve) == amount
```
Comment (rareweasel) [inline tests/functional/test_voting_escrow.py:389]: Question:

How can we assert a migration (with locks, amounts, etc)? I guess we should create a `real` migrator. Right?

Comment (rareweasel) [inline tests/functional/test_voting_escrow.py:389]: Question:

How can we assert a migration (with locks, amounts, etc)? I guess we should create a `real` migrator. Right?

**Review by rareweasel — CHANGES_REQUESTED** [review]
Great job @pandadefi as always!

I left some comments/questions.

File: contracts/Gauge.sol [hunk 5]
```diff
@@ -214,7 +214,7 @@ contract Gauge is BaseGauge, IGauge {
     }
 
     function _lockingRatio(address acccount) internal view returns (uint256) {
-        if (IVotingEscrow(veToken).unlocked()) return PRECISON_FACTOR;
+        if (IVotingEscrow(veToken).migration()) return PRECISON_FACTOR;
```
[the remark this answers is not in the document]
Comment (pandadefi) [reply contracts/Gauge.sol:217]: It's not migrated untill everyone has migrated, that's why it's migration like it's in the process of migration.

File: contracts/VotingEscrow.vy (hunk 1 above)
Comment (pandadefi) [reply contracts/VotingEscrow.vy:436]: `_deposit_for` is doing the check for all deposits, lock extension...

File: contracts/VotingEscrow.vy (hunk 2 above)
Comment (pandadefi) [reply contracts/VotingEscrow.vy:614]: Here the users are the one doing the migration it's not possible to clear the value.

**Pull request merged**